src/components/TroloBoard.tsx

```diff
@@ -0,0 +1,177 @@
+import React, { useMemo, useState } from 'react';
+import { createPortal } from 'react-dom';
+import { useDispatch } from 'react-redux';
+import { useSelector } from 'react-redux';
+import {
+  DndContext,
+  DragEndEvent,
+  DragOverEvent,
+  DragOverlay,
+  DragStartEvent,
+  PointerSensor,
+  useSensor,
+  useSensors,
+} from '@dnd-kit/core';
+import { arrayMove, SortableContext } from '@dnd-kit/sortable';
+
+import { ColumnsSlice, createNewColumn, updateColumnState } from '../redux/columnsSlice';
+import { RowsSlice, updateRowState } from '../redux/rowsSlice';
+import { RootState } from '../redux/store';
+
+import { ColumnContainer } from './dnd/ColumnContainer';
+import { SingleRow } from './dnd/SingleRow';
+
+export const TroloBoard = () => {
+  const dispatch = useDispatch();
+
+  const columns = useSelector((state: RootState) => state.columns);
+  const rows = useSelector((state: RootState) => state.rows);
+  const sensors = useSensors(useSensor(PointerSensor, { activationConstraint: { distance: 10 } }));
+
+  const [activeColumn, setActiveColumn] = useState<ColumnsSlice | null>(null);
+  const [activeRow, setActiveRow] = useState<RowsSlice | null>(null);
+
+  const columnsId = useMemo(() => columns.map((column) => column.id), [columns]);
+  const numberOverRowIndex = 1;
+  const onDragStart = (event: DragStartEvent) => {
+    setActiveColumn(null);
+    setActiveRow(null);
+
+    if (event.active.data.current?.type === 'Column') {
+      setActiveColumn(event.active.data.current.column);
+      return;
+    }
+
+    if (event.active.data.current?.type === 'Row') {
+      setActiveRow(event.active.data.current.row);
+      return;
+    }
+  };
+
+  const onDragEnd = (event: DragEndEvent) => {
+    setActiveColumn(null);
+    setActiveRow(null);
+    const { active, over } = event;
+    if (!over) return;
+
+    const activeId = active.id;
+    const overId = over.id;
+
+    if (activeId === overId) return;
+
+    const getNewColumns = (columns: ColumnsSlice[]) => {
```

Comment:
żeby nie było zagnieżdżania funkcji w funkcji to wrzuć to nad onDragEnd, będzie łatwiej testować potem